tools/online_editor/src/worker/lsp_worker.ts

```diff
@@ -10,57 +10,87 @@ import { InitializeParams, InitializeResult } from "vscode-languageserver";
 
 import slint_init, * as slint_lsp from "@lsp/slint_lsp_wasm.js";
 
-slint_init().then((_) => {
-    const reader = new BrowserMessageReader(self);
-    const writer = new BrowserMessageWriter(self);
+const port_promise = new Promise<MessagePort>((resolve) => {
+    self.onmessage = (event) => {
+        const port = event.ports?.[0];
+        if (port !== null) {
+            resolve(port);
+        }
+    };
+});
 
-    let the_lsp: slint_lsp.SlintServer;
+slint_init()
+    .then(() => port_promise)
+    .then((previewer_port) => {
```

Comment:
can `await` be used  to make the code more readable?

Reply:
can't you do something like

```js
slint_init()
    .then(async () => {
        //.. i can use await here ..
    });
```

Reply:
Why can't we use top-level await?

Reply:
I tried locally - with bumping the output target to es2017 - and it works for me. I can do a follow-up change after this PR to apply it, if you'd like.